File: lab2-functionalprimer/python3/func_as_arg.py
def listFunc():
    int 1
    int 2
    int 3
    int 4
    int 5
    return [i for i in range(1, 6)] #Create list of ints from 1 to 5, Haskell equivalent [1..5]

def applicatorFunc(inpFunc, s):
    if s=='s':
        return sum(inpFunc())
    else:
        return sum(inpFunc())/5

print(applicatorFunc(listFunc, 's'))

File: lab2-functionalprimer/haskell/func_as_arg.hs
--create inpFunc
inpFunc = [a..b] 

--Define applicatorFunc
applicatorFunc inpFunc s = if s=='s' then sum inpFunc else (sum inpFunc)/5  
sgn ::(Ord s, Num s) => s -> Int
sgn x | x < 0 = -1
      | x == 0 = 0
      | x > 0 = 1

main = do
    let result = applicatorFunc inpFunc 'a' --Call applicatorFunc with inpFunc and 'a' as args
    putStrLn("sum = " ++ show(result))